src/client/browser/idle-page/status-indicator.js

```diff
@@ -0,0 +1,234 @@
+        this.rotationAngle            = 0;
+        this.size                     = null;
+        this.spinnerCenter            = null;
+        this.gradient                 = null;
+
+        this._setSize();
+        this._setFontSize();
+        this._setSpinnerGradient();
+
+        StatusIndicator._createStatusMessage(this.connected);
+        StatusIndicator._alignContainerVertically();
+
+        this._drawSpinner(this.connected, 0);
+
+        window.onresize = (() => {
+            var oldSize = this.size;
+
+            this._setSize();
+            this._setFontSize();
+
+            StatusIndicator._alignContainerVertically();
+
+            if (oldSize !== this.size) {
+                if (this.connected)
+                    this._setSpinnerGradient();
+
+                this._drawSpinner(this.connected, this.rotationAngle);
+            }
+
+        });
+    }
+
+
+    //Markup
+    static _getContainer () {
+        return document.getElementsByClassName('container')[0];
+    }
+
+    static _getStatusElementSpan () {
+        return document.getElementsByClassName(STATUS_ELEMENT_CLASS_NAME)[0].children[0];
+    }
+
+    static _createStatusMessage (connected) {
+        var statusSpan = StatusIndicator._getStatusElementSpan();
+
+        statusSpan.className   = connected ? CONNECTED_CLASS_NAME : DISCONNECTED_CLASS_NAME;
+        statusSpan.textContent = connected ? CONNECTED_TEXT : DISCONNECTED_TEXT;
+    }
+
+    static _alignContainerVertically () {
+        var background = document.getElementsByClassName(PAGE_BACKGROUND_CLASS_NAME)[0];
+        var container  = StatusIndicator._getContainer();
+
+        var topMargin = Math.ceil((background.offsetHeight - container.offsetHeight) / 2);
+
+        if (topMargin > 0)
+            container.style.marginTop = convertToString(topMargin);
+    }
+
+
+    _setSize () {
+        var documentElement = window.document.documentElement;
+        var minResolution   = Math.min(documentElement.clientWidth, documentElement.clientHeight);
+        var container       = StatusIndicator._getContainer();
+        var newSize         = Math.round(Math.min(MAXIMUM_SPINNER_SIZE, minResolution * RELATED_SPINNER_SIZE));
+
+        if (newSize === this.size)
+            return;
+
+        this.size          = Math.round(Math.min(MAXIMUM_SPINNER_SIZE, minResolution * RELATED_SPINNER_SIZE));
+        this.spinnerCenter = this.size / 2;
+
+        container.style.width  = convertToString(this.size);
+        container.style.height = convertToString(this.size);
+
+        this.canvas.width = this.canvas.height = this.size;
+    }
+
+    _setFontSize () {
+        var userAgentSpan = document.getElementsByClassName(USER_AGENT_ELEMENT_CLASS_NAME)[0].children[0];
+        var statusSpan    = StatusIndicator._getStatusElementSpan();
+
+        //NOTE: We have established proportions for two edge cases: the maximum spinner size of 400px corresponds to the 16px font,
```

Comment:
Make comment lines to not exceed 60-65 characters, Try make lines of equal length. White space befor `NOTE:`